src/realtime/changes/changes-entity-manager.ts

```diff
@@ -0,0 +1,57 @@
+import Entity from '../../entities/_base'
+import { RealtimeClient } from '..'
+import { ChangeEventHandler, ChangesHandler, ChangeType } from './changes-handler'
+import { isDeletable } from '../../helpers/entity'
+
+export default class ChangesEntityManager<T> implements ChangeEventHandler<T> {
+  private readonly changesHandler: ChangesHandler<T>
+  public types: ChangeType[]
+
+  constructor (
+    private readonly mqtt: RealtimeClient,
+    private readonly entity: Entity<any, T>,
+    types: ChangeType[] = [ChangeType.updated, ChangeType.deleted],
+    private readonly disableDefaultHandler: boolean = false,
+    private readonly customHandler: Partial<ChangeEventHandler<T>> | undefined = undefined) {
+    this.changesHandler = new ChangesHandler<T>(
+      this.mqtt,
+      this,
+      {
+        id: this.entity.id,
+        entityName: this.entity.entityName
+      },
+      types
+    )
+
+    this.types = new Proxy(this.changesHandler.types, {})
+  }
+
+  onCreated (payload: T): void {
+    this.customHandler?.onCreated?.(payload)
```

Comment:
Don't we want the custom handler to replace the default handler? This would execute both and could lead to unexpected results no?

Reply:
there is a boolean called `disableDefaultHandler` that disables the default handler the reason I designed it in this way is that maybe the developer wants to notify the user of changes without holding it off right ? 

for example should a notification Campaign has been updated or deleted when that happens

Reply:
The usual way is to not execute anything if a custom handler is provided. Also the disableDefaultHandler works globally and not only for a specific change handler like `onCreated` no?

Reply:
my issue is that I don't want to force the developer to implement for example the update entity callback which depends on internal functions just so they show some notification to the user let's huddle about this